tests/publiccloud/boottime.pm

```diff
@@ -228,71 +228,25 @@ our $thresholds_by_flavor = {
     },
 };
 
+sub provider_instance {
+    my ($self, $args) = @_;
+    my $provider;
+    my $instance;
 
-sub systemd_time_to_sec
-{
-    my $str = trim(shift);
-    if ($str !~ /^(?<check_min>(?<min>\d{1,2})\s*min\s*)?((?<sec>\d{1,2}\.\d{1,3})s|(?<ms>\d+)ms)$/) {
-        die("Unable to parse systemd time '$str'");
-    }
-    my $sec = $+{sec} // $+{ms} / 1000;
-    $sec += $+{min} * 60 if (defined($+{check_min}));
-    return $sec;
-}
-
-sub extract_analyze {
-    my $string = shift;
-    my $res = {};
-    ($string) = split(/\r?\n/, $string, 2);
-    $string =~ s/Startup finished in\s*//;
-    $string =~ s/=(.+)$/+$1 (overall)/;
-    for my $time (split(/\s*\+\s*/, $string)) {
-        $time = trim($time);
-        my ($time, $type) = $time =~ /^(.+)\s*\((\w+)\)$/;
-        $res->{$type} = systemd_time_to_sec($time);
-    }
-    map { die("Fail to detect $_ timing") unless exists($res->{$_}) } qw(kernel initrd userspace overall);
-    return $res;
-}
-
-sub extract_blame {
-    my $string = shift;
-    my $ret = {};
-    for my $line (split(/\r?\n/, $string)) {
-        $line = trim($line);
-        my ($time, $service) = $line =~ /^(.+)\s+(\S+)$/;
-        $ret->{$service} = systemd_time_to_sec($time);
-    }
-    return $ret;
-}
-
-sub do_systemd_analyze {
-    my ($instance, %args) = @_;
-    $args{timeout} = 120;
-    my $start_time = time();
-    my $output = "";
-    my @ret;
-
-    # Wait for guest register, before calling syastemd-analyze
-    $instance->wait_for_guestregister();
-    while ($output !~ /Startup finished in/ && time() - $start_time < $args{timeout}) {
-        $output = $instance->run_ssh_command(cmd => 'systemd-analyze time', proceed_on_failure => 1);
-        sleep 5;
+    if (get_var('PUBLIC_CLOUD_QAM')) {
+        $provider = $args->{my_provider};
+        $instance = $args->{my_instance};
+    } else {
+        $provider = $self->provider_factory();
```

Comment:
this is only place where we using this variable and just in next line so you may do `$self->provider_factory()->create_instance(check_connectivity => 0);`  or keep it like this if you don't like long lines

Reply:
Possibly, to keep readability of the different functions and classes,  I'd prefer:
 `my $provider =  $self->provider_factory();`

Reply:
ok fine for me

Reply:
See last comment above about boottime.pm